File: 70.climbing-stairs.py
#
# @lc app=leetcode id=70 lang=python3
#
# [70] Climbing Stairs
#

# @lc code=start
class Solution:
    def climbStairs(self, n: int) -> int:
        if n < 3:
            return n

        step_prev1, step_prev2 = 2, 1
        for _ in range(2, n):
            step_prev1, step_prev2 = step_prev1 + step_prev2, step_prev1
        return step_prev1
        
# @lc code=end

File: C/70.climbing-stairs.c
/*
 * @lc app=leetcode id=70 lang=c
 *
 * [70] Climbing Stairs
 */

// @lc code=start
int climbStairs(int n)
{
  if (n <= 3)
  {
    return n;
  }

  int prev1 = 2, prev2 = 1;
  for (int i = 3; i <= n; i++)
  {
    int temp = prev1 + prev2;
    prev2 = prev1;
    prev1 = temp;
  }

  return prev1;
}
// @lc code=end
